src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.TargetSymbolResolver.cs

```diff
@@ -104,7 +103,7 @@ public void Resolve(IList<ISymbol> results)
                     var candidateMembers = containingSymbol.GetMembers(segment);
                     if (candidateMembers.Length == 0)
                     {
-                        return;
+                        return results.ToImmutableAndFree();
```

Comment:
>return results.ToImmutableAndFree(); [](start = 24, length = 36)

Consider using `break;` for each of these with a single `return results.ToImmutableAndFree();` outside the `while`.